### PR TITLE
feat: view client image on the view client page

### PR DESCRIPTION
**Pull request by aashish-ak**
## Description
 - Add `getClientImage` method in `clients.service.ts`
 - Replace the dummy image with the fetched one

## Related issues and discussion
#107 

## Screenshots, if any
clientimg

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

File: src/app/clients/view-client/view-client.component.ts [hunk 1]
```diff
@@ -87,6 +92,19 @@ export class ViewClientComponent implements OnInit, OnDestroy {
       );
   }
 
+  /**
+   * Fetches the image of the client if it exists.
+   * @param id Id of the client whose image is to be fetched
+   */
+  getClientImage(id: any) {
+    this.clientService.getClientImage(id)
```
Comment (luckyman20) [inline src/app/clients/view-client/view-client.component.ts]: Use resolver here.

Comment (aashish-ak) [reply src/app/clients/view-client/view-client.component.ts]: This function will be called in the `getClientId` function, only if the image exists, so that one should be in the resolver. This can't be in the resolver because if the image doesn't exist and this function is called, 404 HTTP error is thrown and the app crashes. @luckyman20

Comment (luckyman20) [reply src/app/clients/view-client/view-client.component.ts]: Oh, yeah, right. I forgot that resolver works asynchronously.

File: src/app/clients/view-client/view-client.component.html [hunk 2]
```diff
@@ -378,7 +378,7 @@ <h3>{{clientInfo?.displayName}}</h3>
             <td mat-cell *matCellDef="let row">
               <i class="fa fa-cloud-download"></i>
               <!-- <i class="p-l-5 fa fa-eye"></i> -->
-              <i class="p-l-5 fa fa-times" (click)="$event.stopPropagation(); deleteClientDocuments(id, row.id)"></i>
```
Comment (abhaychawla) [inline src/app/clients/view-client/view-client.component.html]: There is no change on this line. Please remove this, it should not be present here.

File: src/app/clients/view-client/view-client.component.ts [hunk 3]
```diff
@@ -49,8 +50,10 @@ export class ViewClientComponent implements OnInit, OnDestroy {
   @ViewChild('sharesPaginator') sharesPaginator: MatPaginator;
   @ViewChild(MatSort) sort: MatSort;
 
-  constructor(private route: ActivatedRoute, private clientService: ClientsService,
-              private router: Router, public dialog: MatDialog) {}
+  constructor(private route: ActivatedRoute,
```
Comment (abhaychawla) [inline src/app/clients/view-client/view-client.component.ts]: You are not supposed to refactor the file in this PR. Please remove unrelated changes.

File: src/app/clients/view-client/view-client.component.ts [hunk 4]
```diff
@@ -75,9 +78,11 @@ export class ViewClientComponent implements OnInit, OnDestroy {
         (res => {
           const groupArray = new Array;
           this.clientInfo = res;
-          this.clientInfo.dateOfBirth = this.clientInfo.dateOfBirth.reverse().join('-');
```
Comment (abhaychawla) [inline src/app/clients/view-client/view-client.component.ts]: Unrelated change.

File: src/app/clients/view-client/view-client.component.ts [hunk 5]
```diff
@@ -75,9 +78,11 @@ export class ViewClientComponent implements OnInit, OnDestroy {
...
+          if (this.clientInfo.imagePresent) {
+            this.getClientImage(this.id);
+          }
+          // this.clientInfo.dateOfBirth = this.clientInfo.dateOfBirth.reverse().join('-');
```
Comment (abhaychawla) [inline src/app/clients/view-client/view-client.component.ts]: Unrelated change.

File: src/app/clients/view-client/view-client.component.ts [hunk 6]
```diff
@@ -15,6 +15,7 @@ import { Router } from '@angular/router';
 export class ViewClientComponent implements OnInit, OnDestroy {
   notes: any = undefined;
   id: number = undefined;
+  clientImg: any = 'https://material.angular.io/assets/img/examples/shiba2.jpg';
```
Comment (abhaychawla) [inline src/app/clients/view-client/view-client.component.ts]: Use the user placeholder image instead present in the assets.

File: src/app/clients/view-client/view-client.component.ts (hunk 1 above)
Comment (abhaychawla) [reply src/app/clients/view-client/view-client.component.ts]: Please add a comment:
```
// TODO: Shift into resolver when code will be refactored.
```

File: src/app/clients/view-client/view-client.component.ts [hunk 7]
```diff
@@ -75,9 +78,11 @@ export class ViewClientComponent implements OnInit, OnDestroy {
         (res => {
           const groupArray = new Array;
           this.clientInfo = res;
-          this.clientInfo.dateOfBirth = this.clientInfo.dateOfBirth.reverse().join('-');
+          if (this.clientInfo.imagePresent) {
+            this.getClientImage(this.id);
+          }
+          // this.clientInfo.dateOfBirth = this.clientInfo.dateOfBirth.reverse().join('-');
           this.clientInfo.activationDate = this.clientInfo.activationDate.reverse().join('-');
-
```
Comment (abhaychawla) [inline src/app/clients/view-client/view-client.component.ts]: Unrelated change.

File: src/app/clients/view-client/view-client.component.ts (hunk 5 above)
Comment (aashish-ak) [reply src/app/clients/view-client/view-client.component.ts]: I should suggest that you should keep this change as for some of the clients whose DOB is not present in the db, it throws the error as mentioned in #363 , so when the exception is raised on the component, the rest of the code below this line isn't executed which made it frustrating while debugging the image code i've fetched.

File: src/app/clients/view-client/view-client.component.html (hunk 2 above)
Comment (aashish-ak) [reply src/app/clients/view-client/view-client.component.html]: This line was already present here i just did the identation, should i remove this entire line?

Comment (abhaychawla) [reply src/app/clients/view-client/view-client.component.html]: Keep the line, don't do the indentation.

File: src/app/clients/view-client/view-client.component.ts (hunk 5 above)
Comment (abhaychawla) [reply src/app/clients/view-client/view-client.component.ts]: This error isn't supposed to be resolved as part of the issue you are working on.

**Comment by abhaychawla** [comment]
@aashish-ak If you are still active, please update this PR soon. Otherwise, we might have to close.

**Comment by abhaychawla** [comment]
This PR is no longer relevant as the clients section has been revamped. Closing.
